decisiontree/models.py

```diff
@@ -191,6 +188,9 @@ class Transition(models.Model):
     next_state = models.ForeignKey(
         TreeState, blank=True, null=True, related_name='next_state')
     tags = models.ManyToManyField('Tag', related_name='transitions', blank=True)
+    completion_text = models.CharField(
+        max_length=1000, blank=True, null=True,
+        help_text="Optional message to send when the transition does not have a next state.")
```

Comment:
A transition without a next state doesn't make sense to me. Do you mean we are in a terminal state and there is no transition?

Reply:
The `rapidsms-decisiontree-app` determines whether a tree has reached a terminal state by checking if a transition has a `next_state`: https://github.com/datamade/rapidsms-decisiontree-app/blob/master/decisiontree/app.py#L122 Rather than, say, checking if the `next_state` is a "terminal state." (Indeed, the Transition model [does not require a next_state](https://github.com/datamade/rapidsms-decisiontree-app/blob/master/decisiontree/models.py#L191).)

We could change the models, so that:
(1) the Transition model requires a `next_state`;
(2) an instance of a TreeState could be "terminal" (a boolean field) with a completion text, and the Question field could be blank: https://github.com/datamade/rapidsms-decisiontree-app/blob/master/decisiontree/models.py#L125

Thoughts?

Reply:
It seems like there are two ways to tell if you are in a terminal state.

1. the state doesn't have any transitions
2. the transition doesn't have a next state.

It seems like the second option is kind for having a final interaction with the user?

I think that's kind of confusing, and I would prefer to use only the first mechanism. Is that possible. Is that desirable?

Reply:
Okay - right. I see what you mean by "the state doesn't have any transitions"....the TreeState only knows about the transitions out of it: https://github.com/datamade/rapidsms-decisiontree-app/blob/master/decisiontree/models.py#L149

I can make some changes to this PR to account for these suggestions. I am not entirely sure where to put the "completion text" in the models, however. One solution: instead of a `Question` model, we would just have a `Text` or `Message` (or whatever) model, and the TreeState would have a relation with that model: https://github.com/datamade/rapidsms-decisiontree-app/blob/master/decisiontree/models.py#L131

Reply:
Why can't we have various Terminal states and have the final "completion state" on the transition to the Terminal state?

Reply:
It makes more sense to me to have the completion text available through the TreeState (similar to how it currently [stores a question as a foreign key relation](https://github.com/datamade/rapidsms-decisiontree-app/blob/master/decisiontree/models.py#L131)). In other words, the TreeState points to text - and that text might be a question or a completion message or whatever.